src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java

```diff
@@ -60,19 +76,17 @@ protected Void doInBackground() throws Exception {
 			if (this.stopGenerating) {
 				break;
 			}
-
-			HttpMessage msg = this.httpMessage.cloneRequest();
-
+			HttpMessage msgClone = msg.cloneRequest();
+			Thread.sleep(51);

```

Comment:
I believe the last outstanding item was to make this timing a user controllable option.
Ref: https://github.com/zaproxy/zap-extensions/pull/295#discussion-diff-61220544